### PR TITLE
Add property to run energy consumers at the best time

### PR DESCRIPTION
**Pull request by sstriker**
Implementation suggestion for #54.  This would need to be paired with a PR to HomeAssistant to expose the new property as a sensor.

### CODE REVIEW

**Comment by sstriker** [comment]
@klaasnicolaas shall I go ahead and add tests, or did you have a different direction in mind here?

File: easyenergy/models.py [hunk 1]
```diff
@@ -201,6 +201,28 @@ def timestamp_return_prices(self) -> list[dict[str, float | datetime]]:
         """
         return self.generate_timestamp_list(self.return_prices)
 
+    @property
+    def hours_priced_equal_or_better_usage(self) -> int:
+        """Return the number of hours with the current price or better for usage.
+
+        Returns
+        -------
+            The number of hours with the current price or better for usage.
```
Comment (klaasnicolaas) [inline easyenergy/models.py:210]:
```suggestion
            The number of hours with the current price for usage.
```

File: easyenergy/models.py [hunk 2]
```diff
@@ -201,6 +201,28 @@ def timestamp_return_prices(self) -> list[dict[str, float | datetime]]:
...
+        current = self.current_usage_price or 0
+        return sum(price <= current for price in self.usage_prices.values())
+
+    @property
+    def hours_priced_equal_or_better_return(self) -> int:
+        """Return the number of hours with the current price or better for return.
+
+        Returns
+        -------
+            The number of hours with the current price or better for return.
```
Comment (klaasnicolaas) [inline easyenergy/models.py:221]:
```suggestion
            The number of hours with the current price for return.
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
Go ahead and adjust the tests to test the missing lines 😉

File: easyenergy/models.py (hunk 1 above)
Comment (sstriker) [reply easyenergy/models.py:210]: That wouldn't be quite correct.  It's counting the number of hours that are the current price or below in case of usage.

Comment (klaasnicolaas) [reply easyenergy/models.py:210]: But also `or better` says nothing about what the code does 😉

File: easyenergy/models.py [hunk 3]
```diff
@@ -201,6 +201,28 @@ def timestamp_return_prices(self) -> list[dict[str, float | datetime]]:
         """
         return self.generate_timestamp_list(self.return_prices)
 
+    @property
+    def hours_priced_equal_or_better_usage(self) -> int:
```
Comment (klaasnicolaas) [inline easyenergy/models.py:205]:
```suggestion
    def hours_priced_equal_or_lower_usage(self) -> int:
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: easyenergy/models.py (hunk 1 above)
Comment (sstriker) [reply easyenergy/models.py:210]: Updated to reflect what the code does.

Comment (sstriker) [reply easyenergy/models.py:210]: Resolved in [66f426b](https://github.com/klaasnicolaas/python-easyenergy/pull/55/commits/66f426b2545ca4da5fe4319d80a7cb3e133b3a20)

**Comment by klaasnicolaas** [comment]
Have one last request, your new data feature is still missing in the readme. Could you add to the listed list in [chapter data](https://github.com/klaasnicolaas/python-easyenergy#electricity-prices)?

**Review by klaasnicolaas — APPROVED** [review]

**Pull request merged**